test/unit/services/awsAdapter_spec.js

```diff
@@ -0,0 +1,174 @@
+"use strict";
+
+var AwsAdapter = require("../../../lib/services/awsAdapter");
+var expect     = require("chai").expect;
+var Bluebird   = require("bluebird");
+var Sinon      = require("sinon");
+var AWS        = require("aws-sdk");
+var ec2        = new AWS.EC2();
+
+Bluebird.promisifyAll(ec2);
+
+require("sinon-as-promised")(Bluebird);
+
+describe("The AwsAdapter class ", function () {
+	var DEFAULT_IMAGE_ID      = "ami-d05e75b8";
+	var DEFAULT_INSTANCE_TYPE = "t2.micro";
+
+	var INVALID_IMAGE_ID      = "ami-invalid";
+
+	var VALID_INSTANCE = {
+		ami  : DEFAULT_IMAGE_ID,
+		type : DEFAULT_INSTANCE_TYPE
+	};
+
+	var INVALID_INSTANCE = {
+		ami  : INVALID_IMAGE_ID,
+		type : DEFAULT_INSTANCE_TYPE
+	};
+	describe("creating a new instance with valid ami, type", function () {
```

Comment:
This test case (like several others in this PR) is overly reliant on stub implementations. This test does not actually verify that the AWS adapter correctly creates an instance since it does not check that the AWS SDK is invoked with the expected arguments. This essentially makes this test case a tautology -- it passes because it the stubs return what we told them to.